pandas/core/indexes/datetimelike.py

```diff
@@ -206,7 +206,7 @@ def take(self, indices, axis=0, allow_fill=True, fill_value=None, **kwargs):
             self, indices, axis, allow_fill, fill_value, **kwargs
         )
 
-    @Appender(_shared_docs["searchsorted"])
+    @doc(IndexOpsMixin.searchsorted, klass="Datetime Index")
```

Comment:
All three make good sense to me, and I think `Datetime-like Index` may looks better. What do you think?

When I was updating this, I focused more on using `doc` decorator to replace the `Appender` and `Substitution`. So that I try to avoid changes of other parts. Adding `klass` is because there is a place holder for that and we might not leave it empty. I am not sure about your second question. BTW, how might I know if it published by Sphinx?

Reply:
> BTW, how might I know if it published by Sphinx?

sorry that I wasn't clear. I just meant the published docs at https://dev.pandas.io/docs/index.html

Reply:
> When I was updating this, I focused more on using `doc` decorator to replace the `Appender` and `Substitution`. So that I try to avoid changes of other parts.

sure